imgui/core.pyx

```diff
@@ -3273,6 +3293,475 @@ def drag_int4(
     ), (inout_values[0], inout_values[1], inout_values[2], inout_values[3])
 
 
+def input_text(str label, str value, cimgui.ImGuiInputTextFlags flags=0):
+    """Display text input widget.
+
+    .. visual-example::
+        :auto_layout:
+        :width: 200
+        :height: 100
+
+        text_val = 'Please, type the coefficient here.'
+        imgui.begin("Example: text input")
+        changed, text_val = imgui.input_text('Amount:', text_val)
+        imgui.text('You wrote:')
+        imgui.same_line()
+        imgui.text(text_val)
+        imgui.end()
+
+    Args:
+        label (str): widget label.
+        value (str): textbox value
+        flags: InputText flags. See:
+            :ref:`list of available flags <inputtext-flag-options>`.
+
+    Returns:
+        tuple: a ``(changed, value)`` tuple that contains indicator of
+            textbox state change and the current text contents.
+
+    .. wraps::
+        bool InputText(
+            const char* label,
+            char* buf,
+            size_t buf_size,
+            ImGuiInputTextFlags flags = 0,
+            ImGuiTextEditCallback callback = NULL,
+            void* user_data = NULL
+        )
+    """
+
+    # return changed, bytes_text.decode('utf-8')
+
+    cdef bytes bytes_text = _bytes(value)
+    cdef char inout_text[2056]
+    strcpy(inout_text, bytes_text)
+
+    changed = cimgui.InputText(
+        _bytes(label), inout_text, sizeof(inout_text), flags, NULL, NULL
+    )
+
+    return changed, bytes_text.decode('utf-8')
```

Comment:
We need to return a buffer, so `inout_text ` instead of `bytes_text`. Also I'm not sure about this `decode('utf-8')` this introduces a compatibility issue in Python2.7.

In Python2.7 this function will accept `str` as `value` but returns `unicode`. We need to either force `str` somehow. Maybe right now for consistency we need some opposite implementation of the `_bytes()` function. Maybe `_from_bytes()`?